File: nagasenacs/src/Nagasena/Proc/Grammars/EventTypeElement.cs
using EventDescription = Nagasena.Proc.Common.EventDescription;
using EventDescription_Fields = Nagasena.Proc.Common.EventDescription_Fields;
using EventType = Nagasena.Proc.Common.EventType;
using EventTypeList = Nagasena.Proc.Common.EventTypeList;
using IGrammar = Nagasena.Proc.Common.IGrammar;

namespace Nagasena.Proc.Grammars {

  internal sealed class EventTypeElement : EventType {

    public readonly Grammar ensuingGrammar;

    internal EventTypeElement(int uriId, string uri, int localNameId, string name, EventTypeList eventTypeList, Grammar ensuingGrammar, IGrammar subsequentGrammar) :
      base(uri, name, uriId, localNameId, EVENT_CODE_DEPTH_ONE, eventTypeList, EventType.ITEM_SE, EventDescription_Fields.EVENT_SE, subsequentGrammar) {
      this.ensuingGrammar = ensuingGrammar;
    }

  }

}

File: nagasena/src/org/openexi/proc/grammars/EventTypeElement.java
package org.openexi.proc.grammars;

import org.openexi.proc.common.EventDescription;
import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventTypeList;
import org.openexi.proc.common.IGrammar;

final class EventTypeElement extends EventType {

  public final Grammar ensuingGrammar;

  EventTypeElement(int uriId, String uri, int localNameId, String name, EventTypeList eventTypeList, Grammar ensuingGrammar, IGrammar subsequentGrammar) {
    super(uri, name, uriId, localNameId, EVENT_CODE_DEPTH_ONE, eventTypeList, EventType.ITEM_SE, EventDescription.EVENT_SE, subsequentGrammar);
    this.ensuingGrammar = ensuingGrammar;
  }

}
